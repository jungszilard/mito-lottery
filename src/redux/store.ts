import { configureStore } from "@reduxjs/toolkit"
import intervalReducer from "./slices/intervalSlice"
import lotteryNumberSlice from "./slices/lotteryNumberSlice"

export const store = configureStore({
  reducer: {
    inetrval: intervalReducer,
    lotteryNumber: lotteryNumberSlice,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
