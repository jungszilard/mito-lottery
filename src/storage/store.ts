import { configureStore } from "@reduxjs/toolkit"
import intervalReducer from "./slices/intervalSlice"

export const store = configureStore({
  reducer: {
    inetrval: intervalReducer,
  },
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
