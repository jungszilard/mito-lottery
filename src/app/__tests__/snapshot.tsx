import Home from "@/src/app/page"
import { store } from "@/src/redux/store"
import { render } from "@testing-library/react"
import { Provider } from "react-redux"

it("renders homepage unchanged", () => {
  const { container } = render(
    <Provider store={store}>
      <Home />
    </Provider>
  )
  expect(container).toMatchSnapshot()
})
