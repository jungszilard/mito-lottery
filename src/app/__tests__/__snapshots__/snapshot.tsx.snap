// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders homepage unchanged 1`] = `
<div>
  <div
    data-testid="home-page"
  >
    <p
      class="text-4.5xl font-bold capitalize mb-8"
    >
      Result
    </p>
    <div
      class="flex mb-8"
    >
      <div
        class="flex bg-mito-secondary text-white font-bold py-4.5 px-6 rounded-base"
      >
        <div
          class="mr-6"
        >
          <p
            data-testid="number-of-ticket"
          >
            Number of tickets:
          </p>
          <p
            class="text-sm"
          >
            Years spent:
          </p>
          <p
            class="text-sm"
          >
            Cost of tickets:
          </p>
        </div>
        <div>
          <p>
            0
          </p>
          <p
            class="text-sm"
          >
            0
          </p>
          <p
            class="text-sm"
          >
            0
            ,00 Ft
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex"
    >
      <div
        class="flex border border-solid border-result-border rounded-base"
      >
        <div
          class="border-r border-solid border-result-border w-32 h-18 flex flex-col justify-between items-center py-3"
        >
          <p
            class="font-bold text-xs"
          >
            2 matches
          </p>
          <p
            class="font-extrabold"
          >
            0
          </p>
        </div>
        <div
          class="border-r border-solid border-result-border w-32 h-18 flex flex-col justify-between items-center py-3"
        >
          <p
            class="font-bold text-xs"
          >
            3 matches
          </p>
          <p
            class="font-extrabold"
          >
            0
          </p>
        </div>
        <div
          class="border-r border-solid border-result-border w-32 h-18 flex flex-col justify-between items-center py-3"
        >
          <p
            class="font-bold text-xs"
          >
            4 matches
          </p>
          <p
            class="font-extrabold"
          >
            0
          </p>
        </div>
        <div
          class="w-32 h-18 flex flex-col justify-between items-center py-3"
        >
          <p
            class="font-bold text-xs"
          >
            5 matches
          </p>
          <p
            class="font-extrabold"
          >
            0
          </p>
        </div>
      </div>
    </div>
    <div
      class="flex h-9.5 mb-6 mt-8"
    >
      <p
        class="my-auto w-40"
      >
        Winning numbers:
      </p>
      <div
        class="flex"
      >
        <div
          class="w-8.5 border border-mito-secondary rounded-base flex justify-center items-center ml-4"
        />
        <div
          class="w-8.5 border border-mito-secondary rounded-base flex justify-center items-center ml-4"
        />
        <div
          class="w-8.5 border border-mito-secondary rounded-base flex justify-center items-center ml-4"
        />
        <div
          class="w-8.5 border border-mito-secondary rounded-base flex justify-center items-center ml-4"
        />
        <div
          class="w-8.5 border border-mito-secondary rounded-base flex justify-center items-center ml-4"
        />
      </div>
    </div>
    <div
      class="flex h-9.5 mb-8"
    >
      <p
        class="my-auto w-40"
      >
        Your numbers:
      </p>
      <div
        class="flex"
      >
        <div
          class="w-8.5 border border-mito-secondary rounded-base flex justify-center items-center ml-4"
        />
        <div
          class="w-8.5 border border-mito-secondary rounded-base flex justify-center items-center ml-4"
        />
        <div
          class="w-8.5 border border-mito-secondary rounded-base flex justify-center items-center ml-4"
        />
        <div
          class="w-8.5 border border-mito-secondary rounded-base flex justify-center items-center ml-4"
        />
        <div
          class="w-8.5 border border-mito-secondary rounded-base flex justify-center items-center ml-4"
        />
      </div>
    </div>
    <div
      class="flex items-center mb-8"
    >
      <div
        class="mr-14"
      >
        Play with random numbers:
      </div>
      <button
        class="cursor-pointer w-8 h-8 border border-mito-primary rounded hover:bg-grey-700"
      />
    </div>
    <p>
      Speed
    </p>
    <div
      class=""
    >
      <input
        class="w-full h-2 bg-mito-secondary rounded-lg appearance-none cursor-pointer"
        id="steps-range"
        max="1000"
        min="1"
        step="1"
        type="range"
        value="1"
      />
    </div>
    <div
      class="justify-center items-center flex overflow-x-hidden overflow-y-auto fixed inset-0 z-50 outline-none focus:outline-none"
    >
      <div
        class="relative w-auto my-6 mx-auto max-w-3xl"
      >
        <div
          class="border-0 rounded-lg shadow-lg relative flex flex-col w-full bg-white outline-none focus:outline-none p-6"
        >
          <div
            class="flex items-start justify-between p-5 border-b border-solid border-slate-200 rounded-t"
          >
            <h3
              class="text-3xl font-semibold"
            >
              Pick your lottery numbers
            </h3>
          </div>
          <div
            class="relative p-6 flex-auto"
          >
            <form>
              <div
                class="mb-3 flex justify-around items-center"
              >
                <label>
                  Pick your lottery number 
                  1
                  :
                </label>
                <input
                  class="w-8.5 h-9.5 ml-5 border border-mito-secondary rounded-base text-gray-700 leading-tight focus:outline-none focus:shadow-outline text-center"
                  name="number1"
                  type="number"
                />
              </div>
              <div
                class="mb-3 flex justify-around items-center"
              >
                <label>
                  Pick your lottery number 
                  2
                  :
                </label>
                <input
                  class="w-8.5 h-9.5 ml-5 border border-mito-secondary rounded-base text-gray-700 leading-tight focus:outline-none focus:shadow-outline text-center"
                  name="number2"
                  type="number"
                />
              </div>
              <div
                class="mb-3 flex justify-around items-center"
              >
                <label>
                  Pick your lottery number 
                  3
                  :
                </label>
                <input
                  class="w-8.5 h-9.5 ml-5 border border-mito-secondary rounded-base text-gray-700 leading-tight focus:outline-none focus:shadow-outline text-center"
                  name="number3"
                  type="number"
                />
              </div>
              <div
                class="mb-3 flex justify-around items-center"
              >
                <label>
                  Pick your lottery number 
                  4
                  :
                </label>
                <input
                  class="w-8.5 h-9.5 ml-5 border border-mito-secondary rounded-base text-gray-700 leading-tight focus:outline-none focus:shadow-outline text-center"
                  name="number4"
                  type="number"
                />
              </div>
              <div
                class="mb-3 flex justify-around items-center"
              >
                <label>
                  Pick your lottery number 
                  5
                  :
                </label>
                <input
                  class="w-8.5 h-9.5 ml-5 border border-mito-secondary rounded-base text-gray-700 leading-tight focus:outline-none focus:shadow-outline text-center"
                  name="number5"
                  type="number"
                />
              </div>
              <div
                class="flex items-center justify-end pt-6 mt-4 border-t border-solid border-slate-200 rounded-b"
              >
                <button
                  class="bg-blue-400 hover:bg-blue-500 text-white active:bg-emerald-600 font-bold uppercase text-sm px-6 py-3 rounded shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1 ease-linear transition-all duration-150"
                  type="button"
                >
                  Play with random numbers
                </button>
                <button
                  class="bg-emerald-400 hover:bg-emerald-500 text-white active:bg-emerald-600 font-bold uppercase text-sm px-6 py-3 rounded shadow hover:shadow-lg outline-none focus:outline-none mr-1 mb-1 ease-linear transition-all duration-150"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
    <div
      class="opacity-25 fixed inset-0 z-40 bg-black"
    />
  </div>
</div>
`;
